#!/usr/bin/env python
PACKAGE = "cart2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

mode_enum = gen.enum([gen.const("Stop", int_t, 0, "Stop everything."),
         gen.const("Manual", int_t, 1, "Joystick control."),
         gen.const("GoToPoint", int_t, 2, "Go to a lat-lon point."),
         gen.const("StationKeeping", int_t, 3, "Keep position around lat-lon point."),
         gen.const("CirclePoint", int_t, 4, "Circle a specified lat-lon point."),
         gen.const("HeadingControl", int_t, 5, "Control the heading only."),
         gen.const("HeadingSurge", int_t, 6, "Control the heading with desired surge."),
         gen.const("VTManual", int_t, 7, "Control the heading only.")],"OpMode")

gen.add("OpMode", int_t, 0, "Select operational mode.", 0, edit_method=mode_enum);
gen.add("ManualPoint", bool_t, 0, "Use a specified lat-lon point.",True);
gen.add("PointLat", double_t,0,"The latitude of the point.",0,-90,90);
gen.add("PointLon", double_t,0,"The longitude of the point.",0,-180,180);
gen.add("UseLocal", bool_t, 0, "Use a specified lat-lon point.",True);
gen.add("PointN", double_t,0,"The north point in the local frame.",0,-200,200);
gen.add("PointE", double_t,0,"The east point in the local frame.",0,-200,200);

gen.add("Radius", double_t,0,"The circle radius.",0,-180,180);
gen.add("Surge", double_t,0,"The desired forward surge.",0,0,2);
gen.add("Yaw", double_t,0,"The desired yaw.",0,-3.14,3.14);
gen.add("Launch", bool_t, 0, "Simulate the b-art launch.",False);

exit(gen.generate(PACKAGE, "RemoteControl", "RadioModem"))